### ISSUE TITLE
Release 5.0.1

### ISSUE THREAD

**Issue by StringEpsilon** [ReactTraining/react-router#6739]
Can we release 5.0.1 soon-ish? 

I ask because the license issue of `react-react-context` is blocking the upgrade for me. 

But also because bug reports have slowed down a lot lately. This is probably as stable as it gets until new features get introduced.

The changelog so far would be:

**Bugfixes:**

* Fixed display name for router context consumer and provider (6a99c93)
* Fixed infinite loop caused by `Redirect` in some scenarios (017f692 and 2ce1d32)
* Fixed license issue for react context polyfill (f9849c8, then 992af48)
* Fixed `Redirect` throwing an exception in StaticRouter without context (3ccbd19)
* Fixed regression in `matchPath` that caused an exception for empty paths (7bd1407)
* Fixed page reload when an exception was thrown in `Link onClick` (82ce94c)
* Fixed warning about createRef or useRef values in wrappedComponentRef when using `withRouter()` (56c829b) 
* Fixed withRouter() wrapped component inside NavLink not working properly (a38ef04)
* Fixed CJS build - previously, incorrect exports were emitted (caa9950)

**Changes:**

* Reduced component depth in `withRouter()` HOC. (10d78bb)
* Changed misleading warning when `withRouter` is used outside a `Router` (10d78bb)
* Reduced install size and bundle size by switching to `mini-create-react-context` (992af48)

---

P.S.: Sorry for not using the issue template.

**Comment by StringEpsilon** [comment]
The following PRs could probably get merged too. I don't see any open review comments on them. The first isn't likely to break things and the latter is just a doc update.

* #6623 - Fixes a small edgecase for withRouter() inside a NavLink.
* #6652 - Documentation update removing sections no longer relevant for 5.0.0

All other PRs I checked are either new features or have unresolved comments.

**Comment by timdorr** [comment]
@mjackson Is the release script all good? If so, I can make sure it happens.

**Comment by timdorr** [comment]
BTW, docs fixes don't depend on a release. They go out as soon as they're merged into the `website` branch.

**Comment by wldcordeiro** [comment]
The fixes here would be so nice. That one with the `Redirect` is completely blocking me from being able to update because of the loop. 😦

**Comment by wldcordeiro** [comment]
Sorry for additional noise, this really would be extremely beneficial to get the patches out. We bit the bullet and made the jump to 5.0 for the redirect bug to bite us only after bumping in a couple layers of dependencies and now we're in a bind waiting for either the patches or to do another set of version changes to revert across the dependency chain.

**Comment by StringEpsilon** [comment]
We also had only 19 issues. 17 of them closed as support requests, duplicates and so on. The remaining two are 1 discussion / feature request and 1 that's probably an upstream bug to `history`.

And with the pending overhaul of the roadmap announced in Ryans blogpost, there isn't much in terms of feature development going on either. Only noteworthy development was the overhaul of the `<Link to={function}>` PR by smashercosmo.

@mjackson @timdorr Is there any block we can do something about?

**Comment by timdorr** [comment]
I just need confirmation the automated release scripts will work. I don't want to screw up and cause a false alarm 😬

**Comment by gridviewdesign** [comment]
Ditto on the `<Redirect />` bug. I spent way to much time trying to figure out what I was doing wrong, only to learn that it was a reported bug.  :(

**Comment by alexkrolick** [comment]
So... some of these bugs are rather nasty. Any thoughts of using semantic-release or something to get these fixes shipped? Or again, anything else we can do to help?

**Comment by wldcordeiro** [comment]
@timdorr I'm assuming you're deferring to @mjackson or @ryanflorence

**Comment by timdorr** [comment]
Yep, I need to know from either of them.

**Comment by mjackson** [comment]
Yes, I believe the release script is solid. I used it to release 5.0. I'll cut the release now just in case there's a problem, but it should be good to go.

**Comment by mjackson** [comment]
Looks like the `react-router-native` tests are broken on master?

**Comment by mjackson** [comment]
OK, this should be good to go in 0c9a10d9807b879912f2dff2fbebffe0aa7048ed

Will release as 5.0.1 as soon as https://travis-ci.org/ReactTraining/react-router/builds/541475597 finishes

For the record @timdorr , the process I used was this:

- `lerna version --exact --no-push patch`
- manually inspect the commit to make sure everything is good
- `git push origin master --tags`

That's all any of us should have to do now that the release process is fully automated.

**Comment by mjackson** [comment]
[All done](https://travis-ci.org/ReactTraining/react-router/jobs/541475601)! I'll update the release notes.

**Comment by timdorr** [comment]
Cool, thanks for confirming!

**Comment by travisdahl** [comment]
Not sure if anyone else has seen this issue with `5.0.1` but for me, the dependency `mini-create-react-context` wasn't being installed. I tried deleting packages and reinstalling etc. Strange though because i see the dependency in the router project here...
https://github.com/ReactTraining/react-router/blob/master/packages/react-router/package.json#L46

and i saw that project get added to my yarn lock BUT it didnt show up in the node modules for some strange reason. When i ran my tests I was seeing...
`Cannot find module 'mini-create-react-context' from 'react-router.js'`

installed 'mini-create-react-context' manually and everything worked, but i dont think i should need to do that right?
